lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -297,28 +297,28 @@ class ClassMetadataInfo implements ClassMetadata
     /**
      * READ-ONLY: The names of the parent classes (ancestors).
      *
-     * @var array
+     * @psalm-var list<class-string>
      */
     public $parentClasses = [];
 
     /**
      * READ-ONLY: The names of all subclasses (descendants).
      *
-     * @var array
+     * @psalm-var list<class-string>
      */
     public $subClasses = [];
 
     /**
      * READ-ONLY: The names of all embedded classes based on properties.
      *
-     * @var array
+     * @psalm-var array<string, mixed[]>
```

Comment:
```suggestion
     * @psalm-var array<string, array{class: class-string, columnPrefix: string|false, declaredField: ?string, originalField: ?string, inherited?: class-string, declared?: class-string}>
```
inferred from `\Doctrine\ORM\Mapping\ClassMetadataFactory::addInheritedEmbeddedClasses`
inferred from `\Doctrine\ORM\Mapping\ClassMetadataInfo::mapEmbedded`

----------------

This looks kind of ugly, and probably need to be refactored in ORM3